src/platform/unix/mod.rs

```diff
@@ -655,16 +653,14 @@ struct BackingStore {
 
 impl BackingStore {
     pub fn new(length: usize) -> BackingStore {
-        unsafe {
-            let string = CString::new(TEMP_FILE_TEMPLATE).unwrap();
-            let string_buffer = strdup(string.as_ptr());
-            let fd = mkstemp(string_buffer);
-            assert!(fd >= 0);
-            assert!(maybe_unlink(string_buffer) == 0);
-            libc::free(string_buffer as *mut c_void);
-            assert!(libc::ftruncate(fd, length as off_t) == 0);
-            Self::from_fd(fd)
-        }
+        let count = GLOBAL_SHM_COUNT.fetch_add(1, Ordering::SeqCst);
+        let timestamp = UNIX_EPOCH.elapsed().unwrap();
+        let pid = unsafe { libc::getpid() };
+        let name = CString::new(format!("/ipc-channel-shared-memory.{}.{}.{}",
+                                        count, pid,
+                                        timestamp.as_secs())).unwrap();
```

Comment:
> Why did you drop the nanosecond part? It seems perfectly plausible for a process to exec() (thus resetting the counter while keeping the PID), and then create new shared memory objects all within a single second...

I didn't think about the exec scenario. This is definitely the part of the PR that I'm most unsure about at the moment. I guess if we go back to the pid, counter, second, nanosecond combo I'd be happy with it, but I also started to wonder if some sort of random hash might be better...